src/main/scala/com/microsoft/hyperspace/index/rules/RuleUtils.scala

```diff
@@ -470,17 +467,12 @@ object RuleUtils {
       index: IndexLogEntry,
       originalPlan: LogicalPlan,
       filesAppended: Seq[Path]): LogicalPlan = {
-    // Transform the location of LogicalRelation with appended files.
+    val provider = Hyperspace.getContext(spark).sourceProviderManager
+    // Transform the relation node to include appended files.
     val planForAppended = originalPlan transformDown {
-      case baseRelation @ LogicalRelation(
-            fsRelation @ HadoopFsRelation(location: FileIndex, _, _, _, _, _),
-            baseOutput,
-            _,
-            _) =>
-        val options = Hyperspace
-          .getContext(spark)
-          .sourceProviderManager
-          .partitionBasePath(location)
+      case l: LeafNode if provider.isSupportedRelation(l) =>
+        val relation = provider.getRelation(l)
+        val options = relation.partitionBasePath
```

Comment:
@sezruby Question. Below line 482, do we need to create a tag on the `originalPlan` or can it be just the logical plan for the relation?:
```scala
val newLocation = index.withCachedTag(
          originalPlan,
          IndexLogEntryTags.INMEMORYFILEINDEX_HYBRID_SCAN_APPENDED) {
          new InMemoryFileIndex(spark, filesAppended, options, None)
        }
```
`getCandidateIndexes` uses the relation plan, so I was curious.

Reply:
Yea getCandidateIndexes using a relation plan, but for here I used originalPlan.
It might be better to use the relation plan here as it's more reusable.

Reply:
thanks, we can do that as a follow up (wanted to minimize the behavior change in this PR)